spec/integration/request_controller_spec.rb

```diff
@@ -350,37 +373,55 @@
 
       include_examples 'authority is not subject to FOI law'
 
+      include_examples 'authority is subject to FOI law'
+
     end
 
     context 'marking overdue request as waiting_response' do
 
       let(:classification) { 'waiting_response1' }
 
+      before do
+        time_travel_to(info_request.date_response_required_by + 2.days)
+      end
+
+      after do
+        back_to_the_present
+      end
+
       it 'displays a thank you message post redirect' do
-        time_travel_to(info_request.date_response_required_by + 2.days) do
-          using_session(login(user)) do
-            classify_request(info_request, classification)
-            message = "Thank you! Hope you don't have to wait much longer."
-            # redirect and receive thank you message
-            expect(page).to have_content(message)
-          end
+        using_session(login(user)) do
+          classify_request(info_request, classification)
+          message = "Thank you! Hope you don't have to wait much longer."
+          # redirect and receive thank you message
+          expect(page).to have_content(message)
         end
       end
 
+      include_examples 'authority is not subject to FOI law'
+
+      include_examples 'authority is subject to FOI law'
+
     end
 
     context 'marking very overdue request as waiting_responses' do
 
       let(:classification) { 'waiting_response1' }
 
+      before do
```

Comment:
Why have these been extracted?

As far as I can tell the behaviour is the same

Reply:
Hmm, I could have explained that better in the commit comment. It's to separate out the time travel setup from the process of classifying to allow us to use shared examples. I agree it's pretty clumsy to have it in the `before`/`after` block but I couldn't work out how else to do it.

Reply:
(going to move this into its own commit step for easier explanation/consideration)